zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -185,6 +185,9 @@ public void acceptExternallyAuthenticatedPrincipal(Principal principal) {
 
     /**
      * Accepts an external subject and principal. (Use with caution)
+     * This method is used to propagate an authentication context. For
+     * example when spawing a new thread for an async task, or when
+     * authenticating externally through Kerberos.
      */
     public void acceptExternalSubjectAndPpal(Subject subject,
```

Comment:
So the method already existed? Perhaps if it had had a less "optimised" name it would have been easier to find.